openedx/MITx-edx-integration-tutor.md

```diff
@@ -0,0 +1,199 @@
+
+These steps will also disable the AuthN SSO MFE, so from here on you'll get normal edX authentication screens (if you're not being bounced to MITx Application).
+
+
+1. Get the gateway IP of the `mitxApplication_default` Docker network:
+
+       docker network inspect mitxpro_default | grep Gateway
+
+2. Log into to edX using your superuser account, and make sure your session stays open. Sessions are pretty long-lived so this just means not closing the browser that you started the session in. (Part of this process will involve mostly breaking authentication so it's important that you are able to get into the admin.)
+3. Stop Tutor: `tutor dev stop`
+4. Change into the configuration root for Tutor:
+
+        cd "$(tutor config printroot)"
+
+5. Create a `env/build/openedx/requirements/private.txt` with the required extensions:
+
+       social-auth-mitxpro
+       openedx-companion-auth
+
+6. Edit the `env/apps/openedx/config/lms.env.yml` file and add:
+
+        FEATURES:
+            ALLOW_PUBLIC_ACCOUNT_CREATION: true
+            SKIP_EMAIL_VALIDATION: true
+
+   The `FEATURES` block (should be at the top).
+
+7. Edit the `env/apps/openedx/settings/lms/production.py` and/or `env/apps/openedx/settings/lms/development.py` settings file. (The former is used by a local instance, where the latter is used by both dev and nightly instances.)
+
+   - Add to the end of the file:
+
+      - `THIRD_PARTY_AUTH_BACKENDS = ['social_auth_mitxpro.backends.MITxProOAuth2']`
+      - `REGISTRATION_EXTRA_FIELDS["country"] = "hidden"`
+      - `AUTHENTICATION_BACKENDS.append('social_auth_mitxpro.backends.MITxProOAuth2')`
+      - `SOCIAL_AUTH_OAUTH_SECRETS = {"mitxpro-oauth2": <MIT_Client_Secret> }` - Client Secret that you just copied after `configure_instance` management command
+      - `IDA_LOGOUT_URI_LIST.append('http://{Domain}:{PORT}/logout/')` - there's an existing one of these around like 300 in production.py too.
+
+    - Find and update:
+
+      - `FEATURES['ENABLE_AUTHN_MICROFRONTEND'] = False` (defaults to True)
+
+8. Build a new `openedx` image: `tutor images build openedx-dev` (this will take a long time)
+9. Run a Docker Compse rebuild: `tutor dev dc build` (this should be pretty quick - it's likely not required, just doing it here for safety)
+10. Restart Tutor: `tutor dev start -d` (omit `-d` if you want to watch the logs)
+11. Check your settings. There's a `print_setting` command that you can use to verify everything is set properly:
+
+    - `tutor dev run lms ./manage.py lms print_setting REGISTRATION_EXTRA_FIELDS`
+    - `tutor dev run lms ./manage.py lms print_setting AUTHENTICATION_BACKENDS`
```

Comment:
Something failed at this point. I followed these instructions to this point and didn't see the module (actually, I got an error). Maybe something changed? I did notice something was complaining about wanting new migrations in `third_party_auth` - not sure if that's connected.

Reply:
That is probably `edx-platform` thing